app/src/main/java/com/example/criminalintent/presentation/crime_detail/CrimeFragment.kt

```diff
@@ -167,11 +204,20 @@ class CrimeFragment : Fragment(), FragmentResultListener {
         }
     }
 
-    private fun onActivityResult(result: ActivityResult) {
-        when (result.resultCode) {
-            RESULT_OK -> {
+    private fun onActivityResult(result: ActivityResult, requestCode: Int) {
+        when {
```

Comment:
Interesting when. I find it easier to read if the check if result ok is outside of the when and then the when only checks the `requestCode`